etherman/etherman.go

```diff
@@ -339,6 +344,26 @@ func (etherMan *Client) VerifyGenBlockNumber(ctx context.Context, genBlockNumber
 	return true, nil
 }
 
+// GetL1BlockUpgradeLxLy It returns the block genesis for LxLy before genesisBlock or error
+func (etherMan *Client) GetL1BlockUpgradeLxLy(ctx context.Context, genesisBlock uint64) (uint64, error) {
+	it, err := etherMan.RollupManager.FilterInitialized(&bind.FilterOpts{
+		Start:   1,
+		End:     &genesisBlock,
+		Context: ctx,
+	})
```

Comment:
This won't work in many providers. You must do the query in chunks

Reply:
I have checked with Alchemy and Infura..  which other provider do you think that I have to test it?

Reply:
I have add a param on genesis to by-pass the auto-discover if the provider doesn't support this